src/Raven.Client/Documents/BulkInsert/BulkInsertOperation.cs

```diff
@@ -263,6 +272,62 @@ public BulkInsertOperation(string database, IDocumentStore store, BulkInsertOpti
 
             _generateEntityIdOnTheClient = new GenerateEntityIdOnTheClient(_requestExecutor.Conventions,
                 entity => AsyncHelpers.RunSync(() => _requestExecutor.Conventions.GenerateDocumentIdAsync(database, entity)));
+
+            _streamLock = new AsyncReaderWriterLock();
+            _lastWriteToStream = DateTime.UtcNow;
+            _timer = new Timer(HandleHeartbeat,
+                new WeakReference(this),
+                HeartbeatCheckInterval,
+                HeartbeatCheckInterval);
```

Comment:
maybe Infinite here and then we will not have to change it in the `HandleHearthbeat` ?

Reply:
The last parameter is The time interval between invocations of callback. we can leave it like that and remove the change from HandleHeartbeat.